zrml/prediction-markets/src/lib.rs

```diff
@@ -1564,15 +1562,11 @@ mod pallet {
                     return Err(Error::<T>::OutcomeMismatch.into());
                 }
             }
-            if let OutcomeReport::Scalar(ref inner) = outcome {
-                if let MarketType::Scalar(ref outcome_range) = market.market_type {
-                    ensure!(
-                        inner >= outcome_range.start() && inner <= outcome_range.end(),
-                        Error::<T>::OutcomeOutOfRange
-                    );
-                } else {
-                    return Err(Error::<T>::OutcomeMismatch.into());
-                }
+            if let OutcomeReport::Scalar(_) = outcome {
+                ensure!(
+                    matches!(&market.market_type, MarketType::Scalar(_)),
+                    Error::<T>::OutcomeMismatch
+                );
```

Comment:
Based on the tests I see that the reported outcome is saturated into the min/max value. However, this is an assumption that relies on another part of the code and I have the hunch that it would be better to enforce the saturation within the `report` function.

Reply:
Not by throwing an error, but by saturating it within the code

Reply:
The reported value is not saturated into the market range. And that's fine by me, because (for transparency) the reported outcome should not depend on the market range. Instead, `redeem` saturated the value into the market range when calculating winnings. So I don't think there's any code that depends on the outcome report being within range.

That's why I removed the check completely.

Reply:
I see the argument. From a tech perspective I'd like to saturate it immediately to avoid errors later down the line. From a practical perspective it is important to not modify the reported to not distort the truth (reported by the oracle). I think we can leave it like this and just assume that `redeem` will always respect the saturation.

Reply:
Right now, the tests that verify this behavior aren't really unit tests, but rather integration tests that verify the correct interaction between `report` and `redeem`. Still, I feel fairly safe right now.

When we make the entire approach a bit more object-oriented, maybe we can find a better middle ground.